src/dependency-manager/src/service-config/v1.ts

```diff
@@ -167,13 +167,23 @@ class LivenessProbeV1 extends BaseSpec {
   @IsString({ always: true })
   timeout?: string;
 
-  @IsOptional({ always: true })
+  @ValidateIf(obj => !obj.command, { always: true })
+  @IsDefined({ always: true, message: 'Path and port should be defined if command is not defined.' })
   @IsString({ always: true })
   path?: string;
 
   @IsOptional({ always: true })
   @IsString({ always: true })
   interval?: string;
+
+  @ValidateIf(obj => !obj.path, { always: true })
+  @IsDefined({ always: true, message: 'Command should be defined if path and port are not defined.' })
+  @IsString({ always: true })
+  command?: string;
+
+  @IsOptional({ always: true })
+  @IsNumber(undefined, { always: true })
+  port?: number;
```

Comment:
Talked to David and we decided to kill the api block in the service config. Mind removing that as part of this ticket?

Reply:
If we remove api then we also need to remove a default liveness_probe
1. it doesn't apply if there are multiple interfaces
2. we don't know if an interface is restful